parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -14,26 +14,48 @@ UBUNTU_RELEASE=$(lsb_release -r -s)
 TELEPORTD_PLUGIN_DOWNLOAD_DIR="/opt/teleportd/downloads"
 TELEPORTD_PLUGIN_BIN_DIR="/usr/local/bin"
 KRUSTLET_VERSION="v0.0.1"
+MANIFEST_FILEPATH="/opt/azure/manifest.json"
 
 cleanupContainerdDlFiles() {
     rm -rf $CONTAINERD_DOWNLOADS_DIR
 }
 
 installContainerRuntime() {
-    {{if NeedsContainerd}}
-        echo "in installContainerRuntime - KUBERNETES_VERSION = ${KUBERNETES_VERSION}"
-        if semverCompare ${KUBERNETES_VERSION} "1.22.0"; then
-            CONTAINERD_VERSION="1.5.9"
-            CONTAINERD_PATCH_VERSION="2"
-            installStandaloneContainerd ${CONTAINERD_VERSION} "${CONTAINERD_PATCH_VERSION}"
-            echo "in installContainerRuntime - CONTAINERD_VERION = ${CONTAINERD_VERSION}"
-        else
-            installStandaloneContainerd ${CONTAINERD_VERSION}
-            echo "in installContainerRuntime - CONTAINERD_VERION = ${CONTAINERD_VERSION}"
+{{if NeedsContainerd}}
+    echo "in installContainerRuntime - KUBERNETES_VERSION = ${KUBERNETES_VERSION}"
+    wait_for_file 120 1 /opt/azure/manifest.json # no exit on failure is deliberate, we fallback below.
+
+    local containerd_manifest
+    local stable_containerd
+    local latest_containerd
+    if [ -f "$MANIFEST_FILEPATH" ]; then
+        stable_containerd="$(jq -r .containerd.installed.stable "$MANIFEST_FILEPATH")"
+        latest_containerd="$(jq -r .containerd.installed.latest "$MANIFEST_FILEPATH")"
+    else
+        echo "WARNING: containerd version not found in manifest, defaulting to hardcoded."
+    fi
+
+    # todo(ace): read 1.22 from a manifest and track it against supported versions
+    if semverCompare ${KUBERNETES_VERSION} "1.22.0"; then
+        containerd_version="$(echo "$latest_containerd" | cut -d- -f1)"
+        containerd_patch_version="$(echo "$latest_containerd" | cut -d- -f2)"
```

Comment:
Getting the `stable_containerd` and `latest_containerd` versions and using it in such places is nice!

Reply:
yeah, I'd love to find a way to make it more "static" so that we can catch failures earlier. e.g., this still requires indexing into the manifest. maybe we could have `make generate` parse the manifest and feed raw vars into the bash scripts or something. 

as-is, if the jq parse has typos, you'll have a bad time

Reply:
eh, actually jq parse failures return null, not empty str, which does error out the containerd install...I guess that's good enough since it'll fail e2e as well as vhd pr build